jwst/master_background/expand_to_2d.py

```diff
@@ -166,13 +170,15 @@ def bkg_for_multislit(input, tab_wavelength, tab_background):
         # will detect that those values are out of range (note the `left`
         # argument to np.interp) and set the output to 0.
         wl_array[np.isnan(wl_array)] = -1.
-
+        # flag values outside of background wavelength table
+        mask_limit = (wl_array > max_wave) | (wl_array < min_wave)
+        wl_array[mask_limit] = -1
         # bkg_flux will be a 2-D array, because wl_array is 2-D.
         bkg_flux = np.interp(wl_array, tab_wavelength, tab_background,
                              left=0., right=0.)
 
         background.slits[k].data[:] = bkg_flux.copy()
-
+        background.slits[k].dq[mask_limit] |= dqflags.pixel['DO_NOT_USE']
```

Comment:
Does the ``|=`` operation here work the same as ``numpy.bitwise_or()``? We really do want a bitwise operation, not an operation based on the "total" value of each array element.

Reply:
```
In [2]: import numpy as np

In [8]: a = np.array((0, 1, 2, 4), dtype=np.uint32)

In [9]: b = np.array((0, 0, 1, 1), dtype=np.uint32)

In [10]: a | b
Out[10]: array([0, 1, 3, 5], dtype=uint32)
```

Reply:
https://docs.scipy.org/doc/numpy/reference/generated/numpy.bitwise_or.html

> Computes the bit-wise OR of the underlying binary representation of the integers in the input arrays. This ufunc implements the C/Python operator `|`.

Reply:
And even more robustly:
`In [5]: a=np.array((0,1,2,4,3,6,7),dtype=np.uint32)

In [6]: b=np.array((0,0,0,1,1,1,1),dtype=np.uint32)

In [7]: a | b
Out[7]: array([0, 1, 2, 5, 3, 7, 7], dtype=uint32)
`
shows that for elements that already have a 1 set, it does not add another 1. So it looks good.

Reply:
But I also like the idea of using the explicit ``np.bitwise_or``, just so it's obvious.

Reply:
The only advantage of `np.bitwise_or()` is that if you need to do use some of the `kwargs` that are standard for `numpy` ufuncs, i.e. `axis=`, etc.  So `np.bitwise_or()` adds some extra functionality on top of `|`, but that's it.

Reply:
And
```
In [8]: a = np.array((0, 1, 2, 4), dtype=np.uint32)

In [32]: a | 1
Out[32]: array([1, 1, 3, 5], dtype=uint32)
```